src/main/java/org/hisp/dhis/query/datavalue/DataValueSetQuery.java

```diff
@@ -0,0 +1,116 @@
+package org.hisp.dhis.query.datavalue;
+
+import java.util.*;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+
+import org.hisp.dhis.model.IdScheme;
+
+@Getter
+@Setter
+@Accessors( chain = true )
+public class DataValueSetQuery
+{
+    private final Set<String> dataSets = new HashSet<>();
+
+    private final Set<String> dataElements = new HashSet<>();
+
+    private final Set<String> dataElementGroups = new HashSet<>();
+
+    private final Set<String> orgUnits = new HashSet<>();
+
+    private final Set<String> orgUnitGroups = new HashSet<>();
+
+    private final Set<String> periods = new HashSet<>();
+
+    private String startDate;
+
+    private String endDate;
+
+    private final Set<String> attributeOptionCombos = new HashSet<>();
+
+    private Boolean children;
+
+    private Boolean includeDeleted;
+
+    private String lastUpdated;
+
+    private String lastUpdatedDuration;
+
+    private Integer limit;
+
+    private IdScheme dataElementIdScheme;
+
+    private IdScheme orgUnitIdScheme;
+
+    private IdScheme categoryOptionComboIdScheme;
+
+    private IdScheme attributeOptionComboIdScheme;
+
+    private IdScheme dataSetIdScheme;
+
+    private IdScheme categoryIdScheme;
+
+    private IdScheme categoryOptionIdScheme;
+
+    private IdScheme idScheme;
+
+    private IdScheme inputOrgUnitIdScheme;
+
+    private IdScheme inputDataSetIdScheme;
+
+    private IdScheme inputDataElementGroupIdScheme;
+
+    private IdScheme inputDataElementIdScheme;
+
+    private IdScheme inputIdScheme;
+
+    public static DataValueSetQuery instance()
+    {
+        return new DataValueSetQuery();
+    }
+
+    public DataValueSetQuery addDataElements( String... dataElements )
```

Comment:
Would it be better to use List as opposed to varargs/array? I think in most cases the caller will not be able to enumerate exactly how many data elements to add. A list may be more dynamic, and you can do List.of() if you need specific data elements. Applies to methods below as well.

Reply:
Yes I thought of this. But I wanted to keep it consistent with `addDimension` in [AnalyticsQuery](https://github.com/dhis2/dhis2-java-client/blob/master/src/main/java/org/hisp/dhis/query/analytics/AnalyticsQuery.java#L56). I can add an overloaded method that takes a List to the `AnalyticsQuery` as well (and maybe deprecate the varargs one?).

Reply:
Good point. So it was me that did this in the first place it seems :)